client/src/containers/overview/project-details/bar-chart/index.tsx

```diff
@@ -0,0 +1,73 @@
+interface BarChartProps {
+  total: number;
+  segments: {
+    value: number;
+    label: string;
+    colorClass: string;
+  }[];
+  orientation?: "horizontal" | "vertical";
+}
+
+const BarChart = ({
+  total,
+  segments,
+  orientation = "horizontal",
+}: BarChartProps) => {
+  const getSize = (value: number) => {
+    const percentage = (value / total) * 100;
+    return `${Math.max(percentage, 0)}%`;
+  };
+
+  const formatAmount = (value: number) => {
+    return `${(value / 1_000_000).toFixed(1)}M`;
+  };
```

Comment:
I wouldn't format anything in the component: we don't know if we will have always millions or not. I would move the format function out of the component and format the segment in the place of the implementation and pass it to the this component formatted.